File: OCH_Tao/ABC/A/352.js
const [, X, Y, Z] = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"))[0].split(" ").map(x => parseInt(x, 10));
console.log(Math.min(X, Y) < Z && Z < Math.max(X, Y) ? "Yes" : "No");

File: OCH_Tao/ABC/D/352.py
N,K = map(int,input().split())
P = list(map(int,input().split()))
L = []
cnt = N
for i in range(N):
  L.append(P.index(i+1))
for i in range(N-K):
  X = L[i:i+K]
  cnt = min(max(X)-min(X),cnt)
print(cnt)